sql/binlog.h

```diff
@@ -883,6 +884,23 @@ class MYSQL_BIN_LOG : public TC_LOG {
     mysql_mutex_assert_not_owner(&LOCK_log);
     return atomic_binlog_end_pos;
   }
+
+  my_off_t get_binlog_end_pos_async(bool async_slave) const {
```

Comment:
Any reason why doing this in get_binlog_end_pos() would not be ideal?

Reply:
fixed